### ISSUE TITLE
Need Elastic licenses removed

### ISSUE THREAD

**Issue by jordanoliver23** [opensearch-project/OpenSearch-Dashboards#1130]
**Describe the bug**

It looks there are some libraries with elastic as the license owner still. For example this one: https://github.com/opensearch-project/OpenSearch-Dashboards/blob/1.1/src/plugins/opensearch_dashboards_react/public/code_editor/.storybook/main.js. 

If this is stated as the license for a particular library then we can't build a managed service that leverages this library. It looks like just an issue in cleaning them up but we would be out of compliance if we distribute libraries with this license. 

**Expected behavior**
The correct attribution for each library. 

**OpenSearch Version**
Please list the version of OpenSearch being used.
1.0

**Additional context**
I don't know a way of easily query for attribution. I'd love some suggestions if you have any.

Copied from #1129:
It looks like this file was missed by the license header sweep in #259. The license header should be updated to reflect the correct licensing.

The default config that both of these files export is under the Apache 2.0 license: https://github.com/elastic/kibana/blob/v7.10.2/packages/kbn-storybook/lib/default_config.ts

Files:
* https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/opensearch_dashboards_react/public/code_editor/.storybook/main.js
* https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/embeddable/.storybook/main.js

**Comment by jordanoliver23** [comment]
thank you. @tmarkley , is there any other mentions of this license in the repo now? Meaning did your scan also do another pass across the entire repo?

**Comment by tmarkley** [comment]
@jordanoliver23 yes, in my PR I removed the Elastic License as an allowed license header from the linter so the build will fail if any files use that header. I also did a manual search and didn't find any other instances.

**Comment by jordanoliver23** [comment]
@tmarkley , just wanted to confirm something. Did you remove the library or attribute another license to the library? I would be allowed to just remove the license file only.

**Comment by jordanoliver23** [comment]
Our legal org asked me to follow up with the following question: How did you determine Apache is the correct license? According to https://github.com/elastic/kibana/blob/v7.10.2/src/plugins/embeddable/.storybook/main.js, the correct license is Elastic License Agreement only. Did Elastic agree to relicense it?

**Comment by tmarkley** [comment]
No, there hasn't been an agreement for that, but the history seems very odd to me. Looking at the commits, these two files were moved from x-pack to the main repo plugin directory, but the license header was not changed. Both of these files merely export a default configuration (which [is/was Apache 2.0 code](https://github.com/elastic/kibana/blob/v7.10.2/packages/kbn-storybook/lib/default_config.ts)), so it's not clear why the license wouldn't reflect Apache 2.0 at the time just as the rest of the non-x-pack code did.

@hyandell is there a process that we need to go through for this?

**Comment by jordanoliver23** [comment]
Unless there's an agreement with elastic that anything left in the opensource repo you can do what you want with or attribute apache then there can be a legal issue. You make a good point that it shouldn't be attributed to elastic, rather apache but we can't arbitrarily do this from what I'm told.

**Comment by jordanoliver23** [comment]
@tmarkley , @hyandell , any word on this? Looks like there's two paths forward. Confirmation from elastic that this license was mis-attributed. OR, rebuild it and do a new pull request to get it in the core with the correct associated license. Then we'll be fine. Is that your take? Am I missing anything?

**Comment by jordanoliver23** [comment]
I guess my ask here is if someone can rebuild the library and do this pull request to replace the existing library w/ the appropriate apache license. That would appease our legal org.

**Comment by hyandell** [comment]
Sorry for the delay - as it's an issue of a legal nature, let me discuss internally and review the history of the situation with @tmarkley.

**Comment by jordanoliver23** [comment]
Just saw this @tmarkley 
https://github.com/opensearch-project/OpenSearch-Dashboards/pull/1156. It hasn't been merged yet but why are you doing this?

**Comment by tmarkley** [comment]
@jordanoliver23 I'm rolling back the changes I made immediately after you opened this ticket. I'm writing up a summary about the history of these files and the license headers to review with Henri and others to resolve any potential legal implications.

**Comment by jordanoliver23** [comment]
@hyandell , looking forward to hearing the outcome here.

**Comment by hyandell** [comment]
Sorry for the delay Jordan. As there is a proposal to remove the storybook code, my feeling is that the issue here is going to be solved by that removal. My understanding from Tommy and Kawika is that they're waiting on any feedback to the other proposal before going ahead with the removal soon.

**Comment by jordanoliver23** [comment]
Thanks @hyandell . Yes, removing it would resolve it, or rebuilding it with the appropriate library will as well. Talking to my developers it seemed as though storybook was a fairly essential library and couldn't just be removed but if you find out differently, that's fine. Thanks for your response. If you can update this thread when you know more, that would be great.

**Comment by kavilla** [comment]
> Talking to my developers it seemed as though storybook was a fairly essential library

Hey @jordanoliver23, do you mind providing the insight on storybook and the details here: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/1171.

Seems like a useful experience but as of the forking before licensing change the storybook feature was half-baked.

**Comment by tmarkley** [comment]
These files were removed in #1172.

**Comment by jordanoliver23** [comment]
@kavilla , the problem is with this license. This license can not be associated to the library.
